kolibri/plugins/management/assets/src/actions.js

```diff
@@ -246,16 +246,57 @@ function showClassEditPage(store, classId) {
   );
 }
 
+
 function showClassEnrollPage(store, classId) {
+  store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', PageNames.CLASS_ENROLL_MGMT_PAGE);
-  // need to replace the following with real logic for grabbing data from server-side.
-  store.dispatch('SET_PAGE_STATE', { classId });
-  store.dispatch('CORE_SET_PAGE_LOADING', false);
-  store.dispatch('CORE_SET_ERROR', null);
   store.dispatch('CORE_SET_TITLE', _managePageTitle('Classes'));
+  store.dispatch('CORE_SET_ERROR', null);
+
+  // current facility
+  const facilityPromise = FacilityResource.getCollection().fetch();
+  // all users in facility
+  const userPromise = FacilityUserResource.getCollection().fetch({}, true);
+  // current class
+  const classPromise = ClassroomResource.getModel(classId).fetch();
+  // users in current class
+  const classUsersPromise =
+    FacilityUserResource.getCollection({ member_of: classId }).fetch({}, true);
+
+  ConditionalPromise.all([facilityPromise, userPromise, classPromise, classUsersPromise]).only(
+    samePageCheckGenerator(store),
+    ([facility, facilityUsers, classroom, clasroomUsers]) => {
+      const pageState = {
+        facility: _facilityState(facility[0]),
+        facilityUsers: facilityUsers.map(_userState),
+        clasroomUsers: clasroomUsers.map(_userState),
+        classroom,
+      };
+      store.dispatch('SET_PAGE_STATE', pageState);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+    },
+    error => {
+      coreActions.handleApiError(store, error);
+    }
+  );
 }
 
 
+function enrollUsersInClass(store, classId, users) {
+  return new Promise((resolve, reject) => {
```

Comment:
instead of returning a promise here, can follow the state-modal-control pattern to hand the view control to vuex store.